### PR TITLE
feat(deps): bump keygen

### PR DESCRIPTION
**Pull request by maaslalani**
Bumps keygen to `v0.4.0`

### CODE REVIEW

File: options.go [hunk 1]
```diff
@@ -52,14 +52,13 @@ func WithMiddleware(mw ...Middleware) ssh.Option {
 func WithHostKeyPath(path string) ssh.Option {
 	if _, err := os.Stat(path); os.IsNotExist(err) {
 		dir, f := filepath.Split(path)
```
Comment (aymanbagabas) [inline options.go:54]: Is this line needed?

Comment (maaslalani) [reply options.go:54]: Ah good catch.

File: options.go [hunk 2]
```diff
@@ -51,15 +50,12 @@ func WithMiddleware(mw ...Middleware) ssh.Option {
 // WithHostKeyFile returns an ssh.Option that sets the path to the private.
 func WithHostKeyPath(path string) ssh.Option {
 	if _, err := os.Stat(path); os.IsNotExist(err) {
-		dir, f := filepath.Split(path)
-		n := strings.TrimSuffix(f, "_ed25519")
-		_, err := keygen.NewWithWrite(filepath.Join(dir, n), nil, keygen.Ed25519)
+		_, err := keygen.New(path+"_ed25519", keygen.WithKeyType(keygen.Ed25519), keygen.WithWrite())
```
Comment (maaslalani) [inline options.go:53]: @aymanbagabas Looks like `keygen` trims the suffix so here we are double suffixing? I tried with the examples and it works like this.

Comment (aymanbagabas) [reply options.go:53]: oof you're right that peace let me fix that and push a new patch release xD

File: options.go [hunk 3]
```diff
@@ -50,7 +50,7 @@ func WithMiddleware(mw ...Middleware) ssh.Option {
 // WithHostKeyFile returns an ssh.Option that sets the path to the private.
 func WithHostKeyPath(path string) ssh.Option {
 	if _, err := os.Stat(path); os.IsNotExist(err) {
-		_, err := keygen.New(path+"_ed25519", keygen.WithKeyType(keygen.Ed25519), keygen.WithWrite())
```
Comment (caarlos0) [inline options.go]: I think here it actually needs to append to preserve the previous behavior... 🤔

Comment (caarlos0) [reply options.go]: oh, nvm, you're right

**Pull request merged**